WebDevStudios/wd_s
# Updates ACF Block Defaults

### DESCRIPTION ###
This adds some updates to our global content blocks via class names and a new ID field.

### Class Names
This adds the following class names for background usage:
- `has-background`
- `color-as-background`
- `video-as-background`

(`image-as-background` was already in place)

Often I find that we need to adjust padding when a background element is in place so the content inside of the block doesn't run into the visible edges. If there's no background set, this is generally not an issue since it flows in the context of the page. `has-background` is added when a color, image, or video is in place.

`video-as-background` already existed as a class in `_media.scss` and was the class on the actual video in the ACF block. I've swapped the classes here so the video gets a class of `video-background` and the block gets the `video-as-background` class to stay consistent with the naming of color and image classes.

### Video Updates
I updated the z-index of the `video-background` class to be `-1` so it sits behind the content rather than on top of it.

I also adjusted the global styles for blocks that have a video to prevent the video from breaking out of the parent container.

### Custom ID
On a couple of recent projects we've had to create a content block which is an in-page navigation, or links to anchor tags on the current page. I've added a Custom ID field to the Other Options field group and am setting an ID on each of the content blocks on the page.

By default, with no custom ID set, the ID will be set as the name of the block and the number of its position on the page. So if we're looking at the 4th block on a page and it's a Generic Content block, its default ID would be `generic-content-4`.

### SCREENSHOTS ###
Backend:
![](https://dl.dropbox.com/s/58lv1wy7ctxy6ez/Screenshot%202018-09-21%2011.26.06.jpg?dl=0)

Video Background before overflow fix:
![](https://dl.dropbox.com/s/p80rfdrpcwyfw6q/Screenshot%202018-09-21%2011.27.25.jpg?dl=0)

Video Background after overflow fix:
![](https://dl.dropbox.com/s/9ruh5x5esyvl7ek/Screenshot%202018-09-21%2011.26.34.jpg?dl=0)

### OTHER ###

### STEPS TO VERIFY ###
Create a block and add a custom ID to it.

Create a block with a background element (color, image, or video) and verify that it has the correct classes in place.

Verify that the video does not break out of the container or sit on top of the content in its own block.

### DOCUMENTATION ###
Yes, at the least an updated screenshot of the Other Options panel if merged in.
